### PR TITLE
Feature: maven artifact input for generate and lint goals

### PR DESCRIPTION
**Pull request by sophiejanssens**

### CODE REVIEW

**Review by bartveenstra — CHANGES_REQUESTED** [review]
The example contains backbase specfic examples. such as the bom file. Please use the bom file that is already in the examples.

File: boat-maven-plugin/src/it/example/boat-artifact-input/pom.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bartveenstra) [inline boat-maven-plugin/src/it/example/boat-artifact-input/pom.xml:1]: You must not refer to internal artifacts or specs

**Review by bartveenstra — CHANGES_REQUESTED** [review]
Plus code coverage must be on par :)

File: boat-maven-plugin/src/it/example/boat-generate/pom.xml [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
     <modules>
         <module>angular</module>
-        <module>java-clients</module>
+<!--        <module>java-clients</module>-->
```
Comment (bartveenstra) [inline boat-maven-plugin/src/it/example/boat-generate/pom.xml:25]: Why is this disabled?

File: boat-maven-plugin/src/it/example/boat-multiple-executions/pom.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>generate-docs</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>doc</goal>
+                        </goals>
+                        <configuration>
+                            <inputSpec>${project.basedir}/src/main/resources/contact-manager_contact-manager-service-api-v2.3.0.yaml</inputSpec>
```
Comment (bartveenstra) [inline boat-maven-plugin/src/it/example/boat-multiple-executions/pom.xml:51]: this is an internal API and cannot be used

File: boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java [hunk 4]
```diff
@@ -29,19 +39,19 @@ public void execute() throws MojoExecutionException {
         throw new MojoExecutionException("No OpenAPI specs found in: " + inputSpec);
       }
 
-      for(File f : inputSpecs){
+      for (File f : inputSpecs) {
         inputSpec = f.getPath();
-        output = new File(outPutDirectory.getPath(),f.getName().substring(0,f.getName().lastIndexOf(".")).concat("-docs"));
+        output = new File(outPutDirectory.getPath(), f.getName().substring(0, f.getName().lastIndexOf(".")).concat("-docs"));
```
Comment (bartveenstra) [inline boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java:44]: Output are always docs? Also for client generation?

Comment (sophiejanssens) [reply boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java:44]: The directory input was only wanted for doc goal

Comment (sophiejanssens) [reply boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java:44]: I can change it for others if wanted

File: boat-maven-plugin/src/it/example/boat-generate/pom.xml (hunk 2 above)
Comment (sophiejanssens) [reply boat-maven-plugin/src/it/example/boat-generate/pom.xml:25]: Sorry that was accident

File: boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java (hunk 4 above)
Comment (bartveenstra) [reply boat-maven-plugin/src/main/java/com/backbase/oss/boat/GenerateFromDirectoryDocMojo.java:44]: Ah! I see that now! Sorry. Disregard the previous comment :)

**Comment by bartveenstra** [comment]
@pa314159 Can you also review this?

**Pull request merged**